src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -414,32 +443,27 @@ protected Object invokeDescribable(String symbol, Object _args) {
                 ud = new UninstantiatedDescribable(symbol, null, dargs);
                 margs.put(p.getName(),ud);
 
-                return invokeStep(metaStep, symbol, new NamedArgsAndClosure(margs, args.body));
+                return invokeStep(metaStep, symbol, new NamedArgsAndClosure(margs, args.body, null));
             } catch (Exception e) {
                 throw new IllegalArgumentException("Failed to prepare "+symbol+" step",e);
             }
         }
     }
 
-    private void reportAmbiguousStepInvocation(CpsStepContext context, StepDescriptor d) {
+    private void reportAmbiguousStepInvocation(CpsStepContext context, StepDescriptor d, @Nullable TaskListener listener) {
         Exception e = null;
-        try {
-            TaskListener listener = context.get(TaskListener.class);
-            if (listener != null) {
-                List<String> ambiguousClassNames = StepDescriptor.all().stream()
-                        .filter(sd -> sd.getFunctionName().equals(d.getFunctionName()))
-                        .map(sd -> sd.clazz.getName())
-                        .collect(Collectors.toList());
-                String message = String.format("Warning: Invoking ambiguous Pipeline Step ‘%1$s’ (%2$s). " +
-                        "‘%1$s’ could refer to any of the following steps: %3$s. " +
-                        "You can invoke steps by class name instead to avoid ambiguity. " +
-                        "For example: steps.'%2$s'(...)",
-                        d.getFunctionName(), d.clazz.getName(), ambiguousClassNames);
-                listener.getLogger().println(message);
-                return;
-            }
-        } catch (InterruptedException | IOException temp) {
-            e = temp;
+        if (listener != null) {
+            List<String> ambiguousClassNames = StepDescriptor.all().stream()
+                    .filter(sd -> sd.getFunctionName().equals(d.getFunctionName()))
+                    .map(sd -> sd.clazz.getName())
+                    .collect(Collectors.toList());
+            String message = String.format("Warning: Invoking ambiguous Pipeline Step ‘%1$s’ (%2$s). " +
+                            "‘%1$s’ could refer to any of the following steps: %3$s. " +
+                            "You can invoke steps by class name instead to avoid ambiguity. " +
+                            "For example: steps.'%2$s'(...)",
+                    d.getFunctionName(), d.clazz.getName(), ambiguousClassNames);
+            listener.getLogger().println(message);
+            return;
         }
         LOGGER.log(Level.FINE, "Unable to report ambiguous step invocation for: " + d.getFunctionName(), e);
```

Comment:
```suggestion
        } else {
            LOGGER.log(Level.FINE, "Unable to report ambiguous step invocation for: " + d.getFunctionName());
        }
```

Reply:
updated in 79d255a8427bc41a575bb6f808b355dfaadecb3e